nuxeo-common/src/test/java/org/nuxeo/common/utils/TestTextTemplate.java

```diff
@@ -157,4 +157,39 @@ public void testEscapeVariable() {
         assertEquals("{$$CZjbsiX748UF583qkbinsQ==}", templates.processText("{$$CZjbsiX748UF583qkbinsQ==}"));
     }
 
+    @Test
+    public void testFreetemplateFileNameValidity() {
+        final TextTemplate textTemplate = new TextTemplate();
+        textTemplate.setFreemarkerParsingExtensions("nxftl");
+
+        final File testResouceDirectory = new File(getClass().getClassLoader().getResource("test-nxftl").getPath());
+        final File tmpdir = new File(System.getProperty("java.io.tmpdir"));
+
+        try {
+            textTemplate.processDirectory(new File(testResouceDirectory, "tutu.xml.nxftl"), tmpdir);
+            textTemplate.processDirectory(new File(testResouceDirectory, "tutu.nxftl"), tmpdir);
+        } catch (Exception e) {
+            Assert.fail("Unexpected exception: " + e.getMessage());
```

Comment:
Don't need to explicity fail here, remove try/catch block and make your test method throws Exception.